### ISSUE TITLE
I have an issue with Redis not parsing messages correctly when subscribing to a channel.

### ISSUE THREAD

**Issue by junghyun-kim** [electric-sheep-co/arduino-redis#67]
Hi!
In the startSubscribingNonBlocking method in redis.cpp, when comparing msgVec[0], I get a `RedisMessageUnknownType` error in the following part.

```
if (msgVec[0] != "message" && msgVec[0] != "pmessage"){ 
      emitErr(RedisMessageUnknownType);
      Continue
}
```

I printed out msgVec[0] and saw a number printed like `1) message`,  instead of `message`.
Also, msgVec[1] contains `2) <channel_name>` instead of `<channel_name>`, and msgVec[2] contains `3) <message>`instead of `<message>`.

The beginning of each element in msgVec contains "1)", "2)", and "3)", just like the output from redis-cli. 
Is the problem with this Arduino redis library? Or do I need to modify some settings on my redis server to fix this?

**Comment by rpj** [comment]
@junghyun-kim Thank you for your report, and apologies for the trouble you're having! This is indeed a bug, in fact it's a regression introduced by the changes in the recently-released version 2.6.0. We will address this with the highest priority and will release a version 2.6.1 with the fix as soon as it's available. This ticket will track that work.

@nafraf Admittedly I missed this during code review, but as @junghyun-kim is reporting, you have added the "X) " strings to each value [here](https://github.com/electric-sheep-co/arduino-redis/blob/master/RedisInternal.cpp#L64). The resulting vector should _only_ contain the values stored in Redis without any modifications (including [adding whitespace](https://github.com/electric-sheep-co/arduino-redis/blob/master/RedisInternal.cpp#L87)). Do you think you'd be able to get to this relatively quickly? I'd like to get a fix out ASAP as this will break people's usage. If not, no worries! I should be able to address it tonight.

**Comment by rpj** [comment]
Resolved by #68, published as version 2.6.1 (#69, [tag 2.6.1](https://github.com/electric-sheep-co/arduino-redis/releases/tag/2.6.1)). This release should be available in the Arduino Library Manager at-most 24 hours from now. Thank you again for the report @junghyun-kim and my apologies for your troubles!!

**Comment by rpj** [comment]
@junghyun-kim It just occurred to me that you may still see that "unknown type" error with 2.6.1. If so please file another issue so I can track and fix that separately. Apologies again for the churn but many thanks for helping get this worked out!

**Comment by rpj** [comment]
@junghyun-kim yeah in fact the more I think about it, I'm quite certain you will still see that "unknown type" error because of [this line](https://github.com/electric-sheep-co/arduino-redis/blob/master/RedisInternal.cpp#L90) -- the trouble is I'm not entirely sure of a good way to reproduce it in the tests, so that I can ensure it is fixed properly. If you do see it still with 2.6.1 and don't mind sharing exactly what internal data is causing that (potentially by dumping it directly from `redis-cli`), that would be mighty helpful. Thank you again!!

**Comment by junghyun-kim** [comment]
@rpj Awesome! Thank you for the fast fix and release. I will share the result after the testing. Thank you!!

**Comment by rpj** [comment]
Thank you a ton @junghyun-kim! I'm happy to report that I was able to figure out a way to reproduce the "unknown error" issue too and have just release [version 2.6.2](https://github.com/electric-sheep-co/arduino-redis/releases/tag/2.6.2) with a fix for that as well! So please just skip right over 2.6.1 and give 2.6.2 a try 😄 Thank you again!

**Comment by junghyun-kim** [comment]
Thank you @rpj.
I tested the new version (2.6.2) and it worked as expected! 
I had to use the `-frtti` flag (runtime type information) to use `dynamic_cast` (this issue was already addressed in #72 ).
